impeller/docs/faq.md

```diff
@@ -100,3 +100,113 @@
+    cached at the launch of a Flutter application.
+  * As Flutter applications began to become more complex, the generic shader
+    warmup was no longer suitable for all apps. So the ability to specify
+    shaders to warm up was [exposed to end
+    users](https://api.flutter.dev/flutter/painting/ShaderWarmUp-class.html).
+  * But, this was exceedingly hard to do and also platform specific (iOS Metal,
+    Android OpenGL, Fuchsia Vulkan, etc.). So much so that only a handful of
+    engineers (who all worked on the Flutter Engine) could effectively build the
+    rendering intent that would warm up the right shaders. And these shaders
+    were specific to the app version. If the app was updated to add or remove
+    more features, the warmup routines were invalid and mostly just served to
+    slow down application startup.
+  * To make this process self-serve, [a scheme was
+    devised](https://github.com/flutter/flutter/issues/53607) to run the
+    application and its tests during development in training phase, capture the
+    shaders, package them with the application, and during a new "warm up"
+    phase,  pre-compile the shaders.
+  * But, remember these shaders were platform specific? This moved the burden of
+    platform specfic warmup to the developers. It also made the application
+    sizes a lot bigger since the shaders now had to be packaged with the
+    application. Application launch times were delayed as well.
+  * Even developers who went through these steps weren't always guaranteed
+    freedom from shader compilation jank. If some screens were missed during
+    training runs due to incomplete automated or manual testing, jank could be
+    expected. For complex applications where certain flows were blocked behind
+    authentication or other restrictions, this was an easy situation to get
+    into. Also, some animation frames could be skipped during training because
+    jank during previous frames in the training run. So training runs not only
+    had to be run over the entire app, but also multiple times (on all
+    platforms) to ensure success. Moreover, applications that were really
+    effective at training began to run into egregiously high first-frame times
+    ([some as high as 6
+    seconds](http://github.com/flutter/flutter/issues/97884)).
+  * Due to these reasons, developers abandoned this self-serve warmup mechanism.
+    For instance, no application in Google used this due to the all the problems
+    mentioned.
+  * Flutter began to get a reputation as being hard to write performant apps
+    with. The sort of consistent performance that could be expected from
+    frameworks like UIKit on iOS. Sadly, this was true. And not good enough! We
+    needed to do better.
+  * Impeller was borne out the experience from these events. There were other
+    performance issues the team couldn't address with the previous renderer.
```

Comment:
s/previous/original

and consider:

In additional to the jank issues described above, there were other performance issues inherent in the way Flutter held and used our original renderer.

Reply:
Done.